systemtest/src/test/java/io/strimzi/systemtest/upgrade/StrimziUpgradeIsolatedST.java

```diff
@@ -190,51 +190,14 @@ void testUpgradeAcrossVersionsWithNoKafkaVersion(ExtensionContext extensionConte
         assertNoCoErrorsLogged(0);
     }
 
-    private JsonObject buildDataForUpgradeAcrossVersions() throws IOException {
-        List<TestKafkaVersion> sortedVersions = TestKafkaVersion.getSupportedKafkaVersions();
-        TestKafkaVersion latestKafkaSupported = sortedVersions.get(sortedVersions.size() - 1);
+    private UpgradeDowngradeData getDataForStartUpgrade(List<UpgradeDowngradeData> upgradeYaml) throws IOException {
+        Collections.reverse(upgradeYaml);
 
-        JsonArray upgradeJson = readUpgradeJson(UPGRADE_JSON_FILE);
+        UpgradeDowngradeData startingVersion = null;
 
-        JsonObject acrossUpgradeData = upgradeJson.getJsonObject(upgradeJson.size() - 1);
-        JsonObject startingVersion = getDataForStartUpgrade(upgradeJson);
-
-        acrossUpgradeData.put("fromVersion", startingVersion.getValue("fromVersion"));
-        acrossUpgradeData.put("fromExamples", startingVersion.getValue("fromExamples"));
-        acrossUpgradeData.put("urlFrom", startingVersion.getValue("urlFrom"));
-
-        acrossUpgradeData.put("urlTo", "HEAD");
-        acrossUpgradeData.put("toVersion", "HEAD");
-        acrossUpgradeData.put("toExamples", "HEAD");
-
-        acrossUpgradeData.put("startingKafkaVersion", startingVersion.getString("oldestKafka"));
-        acrossUpgradeData.put("defaultKafka", startingVersion.getString("defaultKafka"));
-        acrossUpgradeData.put("oldestKafka", startingVersion.getString("oldestKafka"));
-
-        // Generate procedures for upgrade
-        JsonObject procedures = new JsonObject();
-        procedures.put("kafkaVersion", latestKafkaSupported.version());
-        procedures.put("logMessageVersion", latestKafkaSupported.messageVersion());
-        procedures.put("interBrokerProtocolVersion", latestKafkaSupported.protocolVersion());
-        acrossUpgradeData.put("proceduresAfterOperatorUpgrade", procedures);
-
-        LOGGER.info("Upgrade Json for the test: {}", acrossUpgradeData.encodePrettily());
-
-        return acrossUpgradeData;
-    }
-
-    private JsonObject getDataForStartUpgrade(JsonArray upgradeJson) throws IOException {
-        List<TestKafkaVersion> sortedVersions = TestKafkaVersion.getSupportedKafkaVersions();
-        Collections.reverse(upgradeJson.getList());
-
-        JsonObject startingVersion = null;
-
-        for (Object item : upgradeJson) {
-            TestKafkaVersion defaultVersion = getDefaultKafkaVersionPerStrimzi(((JsonObject) item).getValue("fromVersion").toString());
-
-            ((JsonObject) item).put("defaultKafka", defaultVersion.version());
-
-            startingVersion = (JsonObject) item;
+        for (UpgradeDowngradeData item : upgradeYaml) {
+            item.setDefaultKafka(item.getDefaultKafkaVersionPerStrimzi());
+            startingVersion = item;
             break;
```

Comment:
Out of curiosity -> how about to remove the for loop, which breaks after one iteration? Also, I would rename the `upgradeYaml` to something like `upgradeDataList`

Reply:
I think this can be part of another PR with upgrade test refactor/rewrite, where it will no longer be needed, as it does return the same object and set only default kafka, which could be done in 2 steps instead of this whole method.

Reply:
Yeah, make sense :)

Reply:
i can do it in this pr as it's simple two step modification and somehow pr related, but further changes in upgrade test processes can be part of another pr